### ISSUE TITLE
Improvement - consider pubkeys using base 58

### ISSUE THREAD

**Issue by orogvany** [semuxproject/semux#728]
If we move to base58, could get a shorter pubkey.

migration would be a problem though, but we'd be able to identify both formats
so probably not too much hassle.

Also would open up wider vanitygen addresses!

**Comment by iiqqrs** [comment]
Yeah orogvany and I were talking about this tonight. I created a vanity gen and realized we don't have much room for cool addresses because of hex. Not anything seriously important.

**Comment by orogvany** [comment]
I can tackle this, and we can add a checksum to the addresses.
We can support both address forms, as 0x denotes the hex, else base58+checksum

**Comment by witoldsz** [comment]
The new address could also start with something predefined, so we could, in the future, introduce new formats, see https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki

Or maybe we should start with the solutions suggested in BIP-0173, like choosing base32 instead of base58:

- Base58 needs a lot of space in QR codes, as it cannot use the alphanumeric mode.
- The mixed case in base58 makes it inconvenient to reliably write down, type on mobile keyboards, or read out loud.
- The double SHA256 checksum is slow and has no error-detection guarantees.
- Most of the research on error-detecting codes only applies to character-set sizes that are a prime power, which 58 is not.
- Base58 decoding is complicated and relatively slow.

What do you think?

**Comment by witoldsz** [comment]
Also, it could be useful to have an account prefixes distinguishing mainnet and testnet, so mainnet address would be rejected by testnet and vice-versa.

**Comment by orogvany** [comment]
Thanks.  Yes a different prefix was what i had in mind too

Base58 conversion in client doesnt need to occur that often, and is still fairly light compared to the other maths we do.  But base 32 works as well.  With Base 32 we can use caps for the checksum like mentioned in other discussion.


Im not sure i buy the typing argument.  Addresses are already terribly long and shortening them would only improve typability. Checksum would resolve error case.  Didnt know about the qr thing. Its odd same number of bits would increase qr size.  Til.
